Classes/DDOSLogger.m

```diff
@@ -79,4 +85,30 @@ - (DDLoggerName)loggerName {
     return DDLoggerNameOS;
 }
 
+- (os_log_t)getLogger {
+    if (self.subsystem == nil || self.category == nil) {
+        return OS_LOG_DEFAULT;
+    }
+    __auto_type subdomain = [self.subsystem UTF8String];
+    __auto_type category = [self.category UTF8String];
+    return os_log_create(subdomain, category);
+}
+
+- (os_log_t)logger {
+    if (_logger == nil)  {
+        _logger = [self getLogger];
+    }
+    return _logger;
+}
+@end
+
+@implementation DDOSLogger (Variations)
+- (instancetype)initWithSubsystem:(NSString *)subsystem category:(NSString *)category {
+    NSAssert(subsystem != nil || category != nil, @"Neither subsytem nor category can be nil. Otherwise, OSLogger falls back to OS_LOG_DEFAULT.");
```

Comment:
This is not completely correct. I can pass nil for both subsystem and category and should get back a valid logger.
```suggestion
    NSAssert((subsystem == nil) == (category == nil), @"Either both subsytem and category or neither can be nil.");
```

Reply:
Also, we should probably override the default initializer (`- init`) and forward it to this one with both parameters being nil.